### PR TITLE
secboot/bootstrap_container.go: also return token writers 

### PR DESCRIPTION
**Pull request by valentindavid**

### CODE REVIEW

**Comment by valentindavid** [comment]
This is needed to store keys in the tokens: See for example https://github.com/canonical/snapd/pull/14608/commits/87ba7340b6099b4ccae38702fdcb88e30cc88ce6

File: secboot/bootstrap_container.go [hunk 1]
```diff
@@ -20,14 +20,16 @@
...
 
 import (
+	"bytes"
 	"fmt"
+	"io"
 
 	"github.com/snapcore/snapd/osutil"
 )
 
 type KeyDataWriter interface {
```
Comment (valentindavid) [inline secboot/bootstrap_container.go:30]: This should have doc

**Review by bboozzoo — APPROVED** [review]
Thanks

File: secboot/bootstrap_container_sb.go [hunk 2]
```diff
@@ -34,6 +34,12 @@ type bootstrappedContainer struct {
 	finished            bool
 }
 
+func newLUKS2KeyDataWriterImpl(devicePath string, name string) (KeyDataWriter, error) {
+	return sb.NewLUKS2KeyDataWriter(devicePath, name)
+}
+
+var newLUKS2KeyDataWriter = newLUKS2KeyDataWriterImpl
+
 func (bc *bootstrappedContainer) AddKey(slotName string, newKey []byte, token bool) (KeyDataWriter, error) {
```
[the remark this answers is not in the document]
Comment (valentindavid) [reply secboot/bootstrap_container_sb.go:43]: In my mind we would be move always use tokens. But after your comment the other day maybe we want to choose to save in tokens or not depending on the version.

[the remark this answers is not in the document]
Comment (valentindavid) [reply secboot/bootstrap_container_sb.go:43]: It would be more "AddKeyAndReturnTokenWriter".

[the remark this answers is not in the document]
Comment (valentindavid) [reply secboot/bootstrap_container_sb.go]: I have pushed a change. Does it look better?

[the remark this answers is not in the document]
Comment (pedronis) [reply secboot/bootstrap_container_sb.go]: yes, functions that have different return values based on bools are a bit strange

[the remark this answers is not in the document]
Comment (valentindavid) [reply secboot/bootstrap_container_sb.go]: I have done a bit more refactoring in another commit. To not have boolean at all even in private methods. If that is OK, I will squash and rebase and make it ready for merge.

**Pull request merged**